src/utils/types.ts

```diff
@@ -77,6 +77,226 @@ export abstract class APIBase {
     }
 }
 
+export class RESTAPI extends APIBase {
+    protected contentType:string;
+    protected acceptType:string;
+
+    get = async (baseurl?:string, contentType?:string, acceptType?:string):Promise<RequestResponseData> => {
+        let ep:string = baseurl ? baseurl : this.baseurl;
+
+        // Set contentType if exists
+        if(contentType) {
+            this.contentType = contentType;
+        }
+
+        // Set acceptType if exists
+        if(acceptType) {
+            this.acceptType = acceptType;
+        }
+
+        let headers:object = {
+            "Content-Type": this.contentType,
+            "Accept": this.acceptType
+        };
+        let axConf:AxiosRequestConfig = {
+            baseURL:this.core.getProtocol()+"://"+this.core.getIP()+":"+this.core.getPort(),
+            responseType: 'json'
+        };
+
+        return this.core.get(ep, {}, headers, axConf).then((resp:RequestResponseData) => {
+            if(resp.status >= 200 && resp.status < 300){
+                if(typeof resp.data === 'object' && 'error' in resp.data) {
+                    throw new Error("Error returned: " + JSON.stringify(resp.data));
```

Comment:
let's remove the throw logic and just return the response. HTTP status codes like 404's have logical meaning in REST. See this guide: https://restfulapi.net/http-status-codes/